packages/drawer/index.tsx

```diff
@@ -184,15 +190,32 @@ class Drawer extends React.Component<DrawerProps, DrawerState> {
   handleKeyDown = (evt: React.KeyboardEvent<HTMLElement>) => {
     this.props.onKeyDown!(evt);
     if (!this.foundation) return;
-    this.foundation.handleKeydown(evt);
+    this.foundation.handleKeydown(evt.nativeEvent);
   };
 
   handleTransitionEnd = (evt: React.TransitionEvent<HTMLElement>) => {
     this.props.onTransitionEnd!(evt);
     if (!this.foundation) return;
-    this.foundation.handleTransitionEnd(evt);
+    this.foundation.handleTransitionEnd(evt.nativeEvent);
   };
 
+  attachRef = (node: HTMLElement) => {
+    const {innerRef} = this.props;
+
+    // https://github.com/facebook/react/issues/13029#issuecomment-410002316
+    // @ts-ignore this is acceptable according to the comment above
+    this.drawerElement.current = node;
+
+    if (innerRef !== undefined) {
```

Comment:
Can we make this an early return? Just do:
```
if (!innerRef) return;
```

Does this need to specifically check for undefined?

Reply:
about the early return: sure! If I check for anything other than undefined I think typescript won't be able to infer that this isn't a "?" type, but I could try